_posts/blog/2018-10-24/2018-10-24-Escrow-Smart-Contract-Specification-in-OpenBazaar.markdown

```diff
@@ -0,0 +1,110 @@
+
+#### Direct Payments
+
+Buyers have the option of _not_ using a moderator when making an OpenBazaar trade. While this isn’t recommended, it may be an acceptable risk for the buyer if the buyer trusts the seller. Direct or unmoderated payments come in two forms: online payments and offline payments.
+
+**Online direct payments** occur when the buyer knows the seller is online. For online payments, the buyer simply sends the funds directly to the sellers wallet after requesting an address to pay to. These are simple, classic transfers of value from one account to another.
+
+**Offline payments** occur when the buyer sees that the seller is offline and is _uncertain_ whether the seller will ever come back online. In this case the buyer sends the funds to a 1-of-2 multisig address with one key held by the buyer and the other held by the seller. If the seller comes back online, they can accept the funds. If the seller doesn’t come back online, the buyer can reclaim the funds.
+
+**Limitations Imposed by OpenBazaar’s Wallet Interface** 
+
+OpenBazaar interacts with all supported coins through its [wallet interface](https://github.com/OpenBazaar/wallet-interface/blob/master/wallet.go#L77). This means that OpenBazaar’s Ethereum smart contracts must be designed in such a way as to be compatible with that interface. OpenBazaar is a live/launched product, so making big changes to the wallet interface in order to support Ethereum is non-trivial. Instead, we’ve decided to keep the wallet interface fixed (for now), and design the smart contract to be compatible with it.
+
+### Intended Use of the Escrow contract
+
+The Escrow contract will store the escrowed funds and state information for _every_ OpenBazaar trade that is using Ethereum (or ERC20 tokens) as the medium of exchange.
+
+We could have, instead, opted to deploy a new escrow contract for each Ethereum-based trade — thereby siloing escrowed funds from each trade in their own smart contract. However, we think the gas requirements for doing so are cost prohibitive, and we fear that would introduce too much friction into Ethereum-based trades.
+
+OpenBazaar trades that use ETH/ERC20 as the medium of exchange are intended to follow the same protocol as those that use a UTXO-based coin as the medium of exchange — and the escrow smart contract is intended to facilitate that.
+
+#### Funding the Trade
+
+Buyers initiate a trade by creating/storing a `Transaction` struct in the Escrow contract and (simultaneously) funding the transaction by sending ETH (or ERC20 tokens) to the Escrow contract. At this point the transaction is in the `FUNDED` state. While in the `FUNDED` state, the buyer may add more ETH (or ERC20 tokens) to escrow if necessary.
+
+#### Releasing Funds from Escrow
+
+While the transaction is in the `FUNDED` state, the escrowed funds can be released only if:
+
+1. Two of the three participants (buyer, seller, and moderator) agree on how the escrowed funds are to be distributed, or 
```

Comment:
End "distributed" with a period. The numbers (1 and 2) act as bullet points, which separate the text into two different sentences.

It would use the comma if the or was inline, with no return before the 2. You'd also want to spell out "One" and "two" if it was a single sentence.